rgw/v2/tests/s3_swift/test_check_sharding_enabled.py

```diff
@@ -0,0 +1,137 @@
+import hashlib
+import json
+import logging
+import time
+import traceback
+
+import v2.lib.resource_op as s3lib
+import v2.utils.utils as utils
+from v2.lib.exceptions import RGWBaseException, TestExecError
+from v2.lib.resource_op import Config
+from v2.lib.rgw_config_opts import CephConfOp, ConfigOpts
+from v2.lib.s3.auth import Auth
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.tests.s3_swift import reusable
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+from v2.utils.utils import RGWService
+
+log = logging.getLogger()
+TEST_DATA_PATH = None
+
+
+def test_exec(config):
+
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    ceph_conf = CephConfOp()
+    rgw_service = RGWService()
+
+    if config.cluster_type == "brownfield":
+        log.info("Check sharding is enabled or not")
+        cmd = "radosgw-admin zonegroup get"
+        out = utils.exec_shell_cmd(cmd)
+        zonegroup = json.loads(out)
+        zonegroup = zonegroup.get("enabled_features")
+        log.info(zonegroup)
+        if "resharding" in zonegroup:
+            log.info("sharding is enabled")
+        else:
+            log.info("sharding is not enabled")
+            log.info("Enabling sharding on cluster since cluster is brownfield")
+            cmd = "radosgw-admin zonegroup get"
+            out = utils.exec_shell_cmd(cmd)
+            zonegroup = json.loads(out)
+            zonegroup_name = zonegroup.get("name")
+            log.info(zonegroup_name)
+            cmd = (
+                "radosgw-admin zonegroup modify --rgw-zonegroup=%s --enable-feature=resharding"
+                % zonegroup_name
+            )
+            out = utils.exec_shell_cmd(cmd)
+            cmd = "radosgw-admin period update --commit"
+            out = utils.exec_shell_cmd(cmd)
+            cmd = "radosgw-admin zonegroup get"
+            out = utils.exec_shell_cmd(cmd)
+            zonegroup = json.loads(out)
+            zonegroup = zonegroup.get("enabled_features")
+            log.info(zonegroup)
+            if "resharding" in zonegroup:
+                log.info("sharding is enabled")
+            else:
+                log.info("sharding has not enabled")
+                test_info.failed_status("test failed")
+                sys.exit(1)
+
+    if config.cluster_type == "greenfield":
+        log.info("Check sharding is enabled or not")
+        cmd = "radosgw-admin zonegroup get"
+        out = utils.exec_shell_cmd(cmd)
+        zonegroup = json.loads(out)
+        zonegroup = zonegroup.get("enabled_features")
+        log.info(zonegroup)
+        if "resharding" in zonegroup:
+            log.info("sharding has enabled already")
+        else:
+            log.info("sharding has not enabled already")
+            test_info.failed_status("test failed")
+            sys.exit(1)
```

Comment:
raise `TestExecError` exception and remove exit. the exception gets caught in the main function and that main function has `exit(1)`